tests/avif/CMakeLists.txt

```diff
@@ -0,0 +1,25 @@
+IF(TARGET avif)
```

Comment:
this looks incorrect.  we write `IF(LIBAVIF_FOUND)` in files instead (like you see below w/png & jpeg).

Reply:
This looks absolutely correct. You should learn modern cmake, read up on targets and interfaces.

https://cliutils.gitlab.io/modern-cmake/chapters/intro/dodonot.html

Several of the "Don'ts" apply to libgd.

Reply:
cmake is not a well supported build system in gd.  our current style is based on FOUND vars, not TARGET.  if we want to rewrite things to use a new pattern, especially if it's one cmake recommends, we can do that, but we should do it across all the build files.  people are just copying & pasting what's already here when writing new code.

keep in mind that we currently support cmake-3.7+.  so any features that require newer versions won't work.

we could prob bump it up to cmake-3.10, but not further at this point (going by Ubuntu Bionic being 3.10).

ftr, i have no real interest in cmake.  i'd just as soon delete it and be done.